docs/contributing.md

```diff
@@ -12,8 +12,49 @@ If you have an idea for a modification or feature, it's probably best to raise a
 
 Everyone contributing to the cibuildwheel project is expected to follow the [PSF Code of Conduct](https://github.com/pypa/.github/blob/main/CODE_OF_CONDUCT.md).
 
-Design Goals
-------------
+## Running the tests
+
+When making a change to the codebase, you can run tests locally for quicker feedback than the CI runs on a PR.
+
+#### Setup
+
+To prepare a development environment, do:
+
+```console
+python3 -m venv .venv
+source .venv/bin/activate
+pip install -e .[dev]
+```
+
+Alternatively, you can have `nox` handle the development environment for you.
+
+#### Unit tests
+
+To run the project's unit tests, do:
+
+```console
+pytest unit_test
+```
+
+There are a few custom options to enable different parts of the test suite - check `pytest unit_test --help` for details.
+
+#### Integration tests
+
+The integration test suite is big - it can take more than 30 minutes to run the whole thing. So normally you'd choose specific tests to run locally, and rely on the project's CI for the rest.
+
+To run them locally, you must pass the platform you want to test via the `CIBW_PLATFORM` environment variable. The `-k` pytest option can be used to select the specific tests. For example, you could do:
```

Comment:
I think it might be reasonable to expose some sort of a CLI option in pytest that would allow autodetecting the current system.

Reply:
I've been thinking more about this and it seems to me that really the root cause is that cibuildwheel refuses to run at all without the `--platform` argument. That was deliberate originally, because it would assume a CI environment and install stuff globally, but we've gradually removed all those bits and now I think it's just a speed bump without much reason. We should consider removing it.

Reply:
Just got that done, it's https://github.com/pypa/cibuildwheel/pull/1727